Remiworks/Remiworks.Attributes/Initialization/Initializer.cs

```diff
@@ -27,10 +27,33 @@ public Initializer(IServiceProvider serviceProvider)
         public void Initialize(Assembly executingAssembly)
         {
             var types = executingAssembly.GetTypes();
+            CheckDependencies(types);
+
             InitializeQueueListeners(types);
             Logger.LogInformation($"Initialization completed. Now listening...");
         }
 
+        private void CheckDependencies(Type[] types)
+        {
+            Logger.LogInformation($"Checking dependencies...");
+            var dependencyChecker = new DependencyInjectionChecker();
+            var errors = dependencyChecker.CheckDependencies(_serviceProvider, types.ToList());
+
+            if (errors.Any())
+            {
+                LogDependencyErrors(errors);
+                throw new DependencyInjectionException("Check the logs for more details!");
```

Comment:
Je zou hier ook een AggregateException kunnen gooien. Dan stop je daar voor iedere class die niet gemaakt kan worden een DependencyInjectionException in met daarin de naam van de class